core/runtime/runtime_api/impl/babe_api.cpp

```diff
@@ -27,4 +27,29 @@ namespace kagome::runtime {
     OUTCOME_TRY(ctx, executor_->ctx().ephemeralAt(block));
     return executor_->call<consensus::babe::Epoch>(ctx, "BabeApi_next_epoch");
   }
+
+  outcome::result<std::optional<consensus::babe::OpaqueKeyOwnershipProof>>
+  BabeApiImpl::generate_key_ownership_proof(
+      const primitives::BlockHash &block_hash,
+      consensus::SlotNumber slot,
+      consensus::babe::AuthorityId authority_id) {
+    OUTCOME_TRY(ctx, executor_->ctx().ephemeralAt(block_hash));
+    return executor_
+        ->call<std::optional<consensus::babe::OpaqueKeyOwnershipProof>>(
+            ctx, "BabeApi_generate_key_ownership_proof", slot, authority_id);
+  }
+
+  outcome::result<void>
+  BabeApiImpl::submit_report_equivocation_unsigned_extrinsic(
+      const primitives::BlockHash &block_hash,
+      consensus::babe::EquivocationProof equivocation_proof,
+      consensus::babe::OpaqueKeyOwnershipProof key_owner_proof) {
+    OUTCOME_TRY(ctx, executor_->ctx().ephemeralAt(block_hash));
+    return executor_->call<void>(
```

Comment:
Must call in "offchain context" (enable tx pool host api).
Here (`submit() { enable(); call(); }`) or before call (`{ enable(); submit(); }`).
Same for other "submit" methods.
```
    offchain_worker_pool_->addWorker(offchain_worker_factory_->make());
    ::libp2p::common::FinalAction remove([&] { offchain_worker_pool_->removeWorker(); });
```